x-pack/plugins/cases/public/components/user_actions/assignees.tsx

```diff
@@ -5,13 +5,159 @@
+  return arrayLength > 1 && index === arrayLength - 1;
+};
+
+const Themselves: React.FC<{
+  index: number;
+  numOfAssigness: number;
+}> = ({ index, numOfAssigness }) => (
+  <FormatListItem index={index} listSize={numOfAssigness}>
+    <>{i18n.THEMSELVES}</>
+  </FormatListItem>
+);
+Themselves.displayName = 'Themselves';
+
+const AssigneeComponent: React.FC<{
+  assignee: Assignee;
+  index: number;
+  numOfAssigness: number;
+}> = ({ assignee, index, numOfAssigness }) => (
+  <FormatListItem index={index} listSize={numOfAssigness}>
+    <UserToolTip profile={assignee.profile}>
+      <strong>{getName(assignee.profile?.user)}</strong>
+    </UserToolTip>
+  </FormatListItem>
+);
+AssigneeComponent.displayName = 'Assignee';
+
+interface AssigneesProps {
+  assignees: Assignee[];
+  currentUserProfile?: UserProfileWithAvatar;
+}
+
+const AssigneesComponent = ({ assignees, currentUserProfile }: AssigneesProps) => (
+  <>
+    {assignees.length > 0 && (
+      <EuiFlexGroup alignItems="center" gutterSize="xs">
+        {assignees.map((assignee, index) => {
+          const usernameDataTestSubj = getUsernameDataTestSubj(assignee);
+
+          return (
+            <EuiFlexItem
+              data-test-subj={`ua-assignee-${usernameDataTestSubj}`}
+              grow={false}
+              key={assignee.uid}
+            >
+              <EuiText size="s" className="eui-textBreakWord">
+                {isCurrentUser(assignee, currentUserProfile) ? (
+                  <Themselves index={index} numOfAssigness={assignees.length} />
+                ) : (
+                  <AssigneeComponent
+                    assignee={assignee}
+                    index={index}
+                    numOfAssigness={assignees.length}
+                  />
+                )}
+              </EuiText>
+            </EuiFlexItem>
+          );
+        })}
+      </EuiFlexGroup>
+    )}
+  </>
+);
+AssigneesComponent.displayName = 'Assignees';
+const Assignees = memo(AssigneesComponent);
+
+const isCurrentUser = (assignee: Assignee, currentUserProfile?: UserProfileWithAvatar) => {
+  return assignee.uid === currentUserProfile?.uid;
+};
+
+const getLabelTitle = (
+  userAction: UserActionResponse<AssigneesUserAction>,
+  userProfiles?: Map<string, UserProfileWithAvatar>,
+  currentUserProfile?: UserProfileWithAvatar
+) => {
+  const assignees = userAction.payload.assignees.map((assignee) => {
+    const profile = userProfiles?.get(assignee.uid);
+    return {
+      uid: assignee.uid,
+      profile,
+    };
+  });
+
+  return (
+    <EuiFlexGroup alignItems="baseline" gutterSize="xs" component="span" responsive={false}>
+      <EuiFlexItem data-test-subj="ua-assignees-label" grow={false}>
+        {userAction.action === Actions.add && i18n.ASSIGNED}
+        {userAction.action === Actions.delete && i18n.UNASSIGNED}
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <Assignees assignees={assignees} currentUserProfile={currentUserProfile} />
```

Comment:
Related to my earlier comment, if the full name isn't set for a user, we need to show the user ID here - currently it is `assigned <empty>`, where it should be `assigned bad_hawk`

<img width="718" alt="image" src="https://user-images.githubusercontent.com/7405507/186673469-81e8680c-cf00-4d00-8f63-a408e19e99ae.png">